overlord/devicestate/handlers_install.go

```diff
@@ -159,6 +164,19 @@ func (m *DeviceManager) doSetupRunSystem(t *state.Task, _ *tomb.Tomb) error {
 		return err
 	}
 
+	// early config
+	ginf, err := gadget.ReadInfo(gadgetDir, nil)
+	if err != nil {
+		return err
+	}
+	defaults := gadget.SystemDefaults(ginf.Defaults)
+	if len(defaults) > 0 {
+		defaultsDir := sysconfig.WritableDefaultsDir(boot.InitramfsWritableDir)
```

Comment:
this the wrong dir btw I think, it should be InstallHostWritableDir. We should probably reorganize the code a bit, let's chat. Initramfs* dirs are for the initramfs (snap-bootstrap) only now.